hack/generator/pkg/codegen/pipeline_convert_allof_and_oneof_to_objects_test.go

```diff
@@ -308,6 +308,8 @@ func TestOneOfResourceSpec(t *testing.T) {
 	g.Expect(err).To(BeNil())
 
 	result := synth.oneOfObject(oneOf, names)
+	result, err = astmodel.TypeAsObjectType(result)
```

Comment:
I thought we were getting rid of th helper method `TypeAsObjectType()` as overpowered and likely to cause subtle bugs.

Reply:
Now that you mention it I do vaguely recall having a discussion about that. There do seem to be some risks, but I like the usage.

Maybe I should instead make a specific `ObjectTypeOrUnwrappedFlaggedType` (name TBD) that has similar usage for my use cases but only works on a singly-nested flagged type (to avoid the "accidentally unwrapped 90 things") problem.